src/Text/Pandoc/Writers/ICML.hs

```diff
@@ -527,11 +537,33 @@ parStyle opts style lst =
       let cont = inTags True "ParagraphStyleRange" attrs' content
       state $ \st -> (cont, st{ blockStyles = Set.insert stlStr $ blockStyles st })
 
+-- | Replace space characters as _
+replaceSpaces :: Text -> Text
+replaceSpaces = Text.concatMap $ \x -> case x of
+  ' ' -> "_"
+  _   -> Text.singleton x
+
+-- | Create the destination name
+makeDestName :: Text -> Text
+makeDestName name = "#" <> replaceSpaces name
```

Comment:
could use [replace](https://www.stackage.org/haddock/lts/text/Data-Text.html#v:replace):

```suggestion
makeDestName name = "#" <> Text.replace " " "_" name
```

Reply:
I was copying a method from later on in the code, so I'll trying replacing that too with replace - perfect!